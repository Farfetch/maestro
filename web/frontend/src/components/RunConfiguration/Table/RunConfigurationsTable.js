import { Col, Row, Space, Table } from "antd";
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";

const columns = [
  {
    title: "Title",
    dataIndex: "title",
    key: "title"
  },
  {
    title: "Created",
    dataIndex: "createdAt",
    key: "createdAt",
    sorter: {
      compare: (recordA, recordB) => recordA.createdAt.diff(recordB.createdAt),
      multiple: 1
    },
    render: (text, record) => record.createdAt.format("L HH:mm:ss"),
    defaultSortOrder: "descend",
    width: 180
  },
  {
    title: "Updated",
    dataIndex: "updatedAt",
    key: "updatedAt",
    sorter: {
      compare: (recordA, recordB) => recordA.updatedAt.diff(recordB.updatedAt),
      multiple: 1
    },
    render: (text, record) => record.updatedAt.format("L HH:mm:ss"),
    width: 180
  },
  {
    title: "Action",
    key: "action",
    width: 200,
    render: (text, record) => (
      <>
        <Space size="middle">
          <Link to={`/test/${record.key}`}>Open</Link>
        </Space>
      </>
    )
  }
];

const runConfigurationMapper = ({
  id,
  title,
  runPlanId,
  createdAt,
  updatedAt
}) => ({
  key: id,
  title,
  runPlanId,
  createdAt,
  updatedAt
});

const RunConfigurationsTable = ({ runConfigurations, isLoading = false }) => {
  const dataSource = runConfigurations.map(runConfigurationMapper);
  const pagination = {
    defaultPageSize: 50,
    hideOnSinglePage: true,
    pageSizeOptions: [50, 100, 200, 1000],
    position: ["bottomRight"]
  };

  return (
    <Row gutter={[32, 32]} justify="start" align="middle">
      <Col span={24}>
        <Table
          size="small"
          loading={isLoading}
          dataSource={dataSource}
          columns={columns}
          pagination={pagination}
        />
      </Col>
    </Row>
  );
};

RunConfigurationsTable.propTypes = {
  runConfigurations: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      runPlanId: PropTypes.string.isRequired,
      createdAt: PropTypes.object.isRequired,
      updatedAt: PropTypes.object.isRequired
    })
  ),
  isLoading: PropTypes.bool.isRequired
};

export default RunConfigurationsTable;
