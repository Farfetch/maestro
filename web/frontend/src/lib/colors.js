import {
  blue,
  cyan,
  geekblue,
  gold,
  green,
  grey,
  lime,
  magenta,
  orange,
  purple,
  red,
  volcano,
  yellow
} from "@ant-design/colors";

export const colors = {
  red,
  volcano,
  orange,
  gold,
  yellow,
  lime,
  green,
  cyan,
  blue,
  geekblue,
  purple,
  magenta,
  grey
};

export const defaultColorsList = [
  blue,
  geekblue,
  purple,
  magenta,
  cyan,
  red,
  volcano,
  orange,
  gold,
  yellow,
  lime,
  green
];

export const getNextColor = (colorIndex, colorsList = defaultColorsList) => {
  const color = colorsList[colorIndex];
  let nextColorIndex = colorIndex;

  if (colorsList[colorIndex + 1]) {
    nextColorIndex += 1;
  } else {
    nextColorIndex = 0;
  }

  return { colorIndex: nextColorIndex, color };
};
