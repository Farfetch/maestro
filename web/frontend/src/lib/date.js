import moment from "moment";

const defaultDatetimeFormat = "YYYY-MM-DD HH:mm:ss";

const hourMinuteFormat = "HH:mm";

export const toUtcString = (localDatetime) =>
  localDatetime.utc().format(defaultDatetimeFormat);

export const toLocalDate = (utcDatetimeStr) =>
  moment.utc(utcDatetimeStr).local();

export const toUtcHourMinute = (localDatetime) =>
  localDatetime.utc().format(hourMinuteFormat);

export const toLocalHourMinute = (utcHourMinute) =>
  moment.utc(utcHourMinute, hourMinuteFormat).local();
