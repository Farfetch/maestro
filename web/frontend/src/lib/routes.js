import { maestroApiUrl } from "../config";

export const runPlanDownloadUrl = (runPlanId) =>
  `${maestroApiUrl}/api/run_plan/${runPlanId}/download`;

export const customDataDownloadUrl = (customDataId) =>
  `${maestroApiUrl}/api/custom_data/${customDataId}/download`;

export const runMetricsDownloadUrl = (runId) =>
  `${maestroApiUrl}/api/run_metrics/${runId}/download`;

export const runLogDownloadUrl = (runId) =>
  `${maestroApiUrl}/api/run_log/${runId}/download`;

export const testSingleUrl = (runConfigurationId) =>
  `/test/${runConfigurationId}`;

export const testNewUrl = "/tests/new";

export const agentLogsUrl = (runConfigurationId) =>
  `/agent/${runConfigurationId}/logs`;

export const runSingleUrl = (runId) => `/run/${runId}`;

export const historyUrl = "/history";

export const logoutUrl = `${maestroApiUrl}/logout`;
export const workspacesUrl = "/workspaces";
export const usersUrl = "/users";
