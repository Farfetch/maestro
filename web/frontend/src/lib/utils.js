export const avg = (value, itemsCount) =>
  itemsCount === 0 ? 0 : value / itemsCount;

export const calculateErrorRate = (successCount, totalCount) => {
  const errorRate = parseFloat((1 - successCount / totalCount) * 100).toFixed(
    2
  );

  return errorRate === "NaN" ? 0 : errorRate;
};
