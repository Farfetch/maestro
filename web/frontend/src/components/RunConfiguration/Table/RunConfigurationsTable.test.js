import { render } from "@testing-library/react";
import MockDate from "mockdate";
import moment from "moment";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { toLocalDate } from "../../../lib/date";
import RunConfigurationsTable from "./RunConfigurationsTable";

describe("components/RunConfiguration/Table", () => {
  afterEach(() => {
    MockDate.reset();
  });
  const runConfigurations = [
    {
      id: "run-configuration-id-1",
      title: "Confgiuration title",
      runPlanId: "run-plan-id",
      createdAt: moment("2018-05-24T13:48:04.313000"),
      updatedAt: moment("2018-05-24T13:48:04.313000")
    }
  ];
  test(`should render RunConfigurationTable component`, async () => {
    MockDate.set(toLocalDate("2022-03-03 00:00:00"));
    const rendered = render(
      <MemoryRouter initialEntries={["/tests"]}>
        <RunConfigurationsTable
          runConfigurations={runConfigurations}
          isLoading={false}
        />
      </MemoryRouter>
    );

    const runConfigurationTableComponent = rendered.container;
    expect(runConfigurationTableComponent.outerHTML).toMatchSnapshot();
  });

  test(`should render RunConfigurationTable loader`, async () => {
    MockDate.set(toLocalDate("2022-03-03 00:00:00"));
    const rendered = render(
      <MemoryRouter initialEntries={["/tests"]}>
        <RunConfigurationsTable
          runConfigurations={runConfigurations}
          isLoading={true}
        />
      </MemoryRouter>
    );

    const runConfigurationTableComponent = rendered.container;
    expect(runConfigurationTableComponent.outerHTML).toMatchSnapshot();
  });

  test(`should render RunConfigurationTable nextRun column`, async () => {
    MockDate.set(toLocalDate("2022-03-03 00:00:00"));
    const runConfigurationsWithSchedule = [
      {
        id: "run-configuration-id-1",
        title: "Confgiuration title",
        runPlanId: "run-plan-id",
        schedule: {
          days: ["Thu"],
          time: "13:50"
        },
        createdAt: moment("2018-05-24T13:48:04.313000"),
        updatedAt: moment("2018-05-24T13:48:04.313000")
      },
      {
        id: "run-configuration-id-2",
        title: "Confgiuration title",
        runPlanId: "run-plan-id",
        schedule: {
          days: ["Thu"],
          time: "13:00"
        },
        createdAt: moment("2018-05-24T13:48:04.313000"),
        updatedAt: moment("2018-05-24T13:48:04.313000")
      },
      {
        id: "run-configuration-id-3",
        title: "Confgiuration title",
        runPlanId: "run-plan-id",
        schedule: {
          days: ["Sun"],
          time: "10:00"
        },
        createdAt: moment("2018-05-24T13:48:04.313000"),
        updatedAt: moment("2018-05-24T13:48:04.313000")
      }
    ];
    const rendered = render(
      <MemoryRouter initialEntries={["/tests"]}>
        <RunConfigurationsTable
          runConfigurations={runConfigurationsWithSchedule}
          isLoading={false}
        />
      </MemoryRouter>
    );

    const runConfigurationTableComponent = rendered.container;
    expect(runConfigurationTableComponent.outerHTML).toMatchSnapshot();
  });
});
