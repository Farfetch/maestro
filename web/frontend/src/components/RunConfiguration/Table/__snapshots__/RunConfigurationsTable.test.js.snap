// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Table should render RunConfigurationTable component 1`] = `"<div><div class=\\"ant-row ant-row-start ant-row-middle\\" style=\\"margin: -16px -16px -16px -16px;\\"><div style=\\"padding: 16px 16px 16px 16px;\\" class=\\"ant-col ant-col-24\\"><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup><col><col style=\\"width: 180px;\\"><col style=\\"width: 180px;\\"><col style=\\"width: 200px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell\\">Title</th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Next Run</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-sort ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Created</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\">Action</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"run-configuration-id-1\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell\\">Confgiuration title</td><td class=\\"ant-table-cell\\"></td><td class=\\"ant-table-cell ant-table-column-sort\\">05/24/2018 13:48</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\"><a href=\\"/test/run-configuration-id-1\\">Open</a></div></div></td></tr></tbody></table></div></div></div></div></div></div></div></div></div>"`;

exports[`components/RunConfiguration/Table should render RunConfigurationTable loader 1`] = `"<div><div class=\\"ant-row ant-row-start ant-row-middle\\" style=\\"margin: -16px -16px -16px -16px;\\"><div style=\\"padding: 16px 16px 16px 16px;\\" class=\\"ant-col ant-col-24\\"><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div><div class=\\"ant-spin ant-spin-spinning\\"><span class=\\"ant-spin-dot ant-spin-dot-spin\\"><i class=\\"ant-spin-dot-item\\"></i><i class=\\"ant-spin-dot-item\\"></i><i class=\\"ant-spin-dot-item\\"></i><i class=\\"ant-spin-dot-item\\"></i></span></div></div><div class=\\"ant-spin-container ant-spin-blur\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup><col><col style=\\"width: 180px;\\"><col style=\\"width: 180px;\\"><col style=\\"width: 200px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell\\">Title</th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Next Run</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-sort ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Created</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\">Action</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"run-configuration-id-1\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell\\">Confgiuration title</td><td class=\\"ant-table-cell\\"></td><td class=\\"ant-table-cell ant-table-column-sort\\">05/24/2018 13:48</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\"><a href=\\"/test/run-configuration-id-1\\">Open</a></div></div></td></tr></tbody></table></div></div></div></div></div></div></div></div></div>"`;

exports[`components/RunConfiguration/Table should render RunConfigurationTable nextRun column 1`] = `"<div><div class=\\"ant-row ant-row-start ant-row-middle\\" style=\\"margin: -16px -16px -16px -16px;\\"><div style=\\"padding: 16px 16px 16px 16px;\\" class=\\"ant-col ant-col-24\\"><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: auto;\\"><colgroup><col><col style=\\"width: 180px;\\"><col style=\\"width: 180px;\\"><col style=\\"width: 200px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell\\">Title</th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Next Run</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-sort ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Created</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\">Action</th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"run-configuration-id-1\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell\\">Confgiuration title</td><td class=\\"ant-table-cell\\">03/03/2022 13:50</td><td class=\\"ant-table-cell ant-table-column-sort\\">05/24/2018 13:48</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\"><a href=\\"/test/run-configuration-id-1\\">Open</a></div></div></td></tr><tr data-row-key=\\"run-configuration-id-2\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell\\">Confgiuration title</td><td class=\\"ant-table-cell\\">03/03/2022 13:00</td><td class=\\"ant-table-cell ant-table-column-sort\\">05/24/2018 13:48</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\"><a href=\\"/test/run-configuration-id-2\\">Open</a></div></div></td></tr><tr data-row-key=\\"run-configuration-id-3\\" class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell\\">Confgiuration title</td><td class=\\"ant-table-cell\\">03/06/2022 10:00</td><td class=\\"ant-table-cell ant-table-column-sort\\">05/24/2018 13:48</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\"><a href=\\"/test/run-configuration-id-3\\">Open</a></div></div></td></tr></tbody></table></div></div></div></div></div></div></div></div></div>"`;
